src/mongoClusters/utils/connectionStringHelpers.ts

```diff
@@ -30,8 +30,41 @@ export const getPasswordFromConnectionString = (connectionString: string): strin
     return new ConnectionString(connectionString).password;
 };
 
+export const getHostsFromConnectionString = (connectionString: string): string[] => {
+    return new ConnectionString(connectionString).hosts;
+};
+
 export const addDatabasePathToConnectionString = (connectionString: string, databaseName: string): string => {
     const connectionStringOb = new ConnectionString(connectionString);
     connectionStringOb.pathname = databaseName;
     return connectionStringOb.toString();
 };
+
+/**
+ * Checks if any of the given hosts end with any of the provided suffixes.
+ *
+ * @param hosts - An array of host strings to check.
+ * @param suffixes - An array of suffixes to check against the hosts.
+ * @returns True if any host ends with any of the suffixes, false otherwise.
+ */
+function hostsEndWithAny(hosts: string[], suffixes: string[]): boolean {
+    return hosts.some((host) => {
+        const hostWithoutPort = host.split(':')[0].toLowerCase();
```

Comment:
Just a question. The URI allows the next string `http://username: password@example.com:8888`
Are you sure that host here will be always `<host>:<port>`? If yes, just ignore

Reply:
Yes, that's my expectation as I'm using a parser from 'mongodb-connection-string-url' and it extracts the host info for me. I'm just not sure about how it handles ports, therefore, avoiding them in case they're in there.